canarytokens/utils.py

```diff
@@ -78,20 +78,24 @@ def get_deployed_commit_sha(commit_sha_file: Path = Path("/COMMIT_SHA")):
 #     return inner
 
 
-def get_src_ip_continent(country: str) -> str:
+def get_src_ip_continent(geo_data: dict) -> str:
     """Helper function that returns the continent of country given it's ISO 3166-2 code.
 
     Args:
-        country (str): ISO 3166-2 code
+        geo_data (dict): The "country" key contains an ISO 3166-2 code
 
     Returns:
         str: A two character code representing a continent
     """
-    # AQ is the ISO 3166-2 code for Antarctica, and is returned from IPinfo,
-    # but it's not included in pycountry_convert.
-    if country == "AQ":
-        return "AN"
-    try:
-        return pycountry_convert.country_alpha2_to_continent_code(country)
-    except KeyError:
+    country = geo_data.get("country")
+    if country is not None:
+        # AQ is the ISO 3166-2 code for Antarctica, and is returned from IPinfo,
+        # but it's not included in pycountry_convert.
+        if country == "AQ":
+            return "AN"
+        try:
+            return pycountry_convert.country_alpha2_to_continent_code(country)
+        except KeyError:
+            return "NO_CONTINENT"
+    else:
         return "NO_CONTINENT"
```

Comment:
Returning early reduces indentation and makes it easier to follow.

```suggestion
    country = geo_data.get("country")
    if country is None:
        return "NO_CONTINENT"
    
    # AQ is the ISO 3166-2 code for Antarctica, and is returned from IPinfo,
    # but it's not included in pycountry_convert.
    if country == "AQ":
        return "AN"
    try:
        return pycountry_convert.country_alpha2_to_continent_code(country)
    except KeyError:
        return "NO_CONTINENT"
```